### PR TITLE
Mention list

### PR DESCRIPTION
**Pull request by iqmulator**

### CODE REVIEW

File: src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionList.razor [hunk 1]
```diff
@@ -1,36 +1,50 @@
...
     var mentions = Mentions;
+    var listCls = "";
     if (mentions.Length < 1)
-        return;
+        listCls += "hidden";
     var selection = Selection;
 }
 
-<div class="@Class mb-1 p-1 border border-mention rounded-md overflow-y-auto max-h-96 w-full">
+<div @ref="Ref" class="@Class @listCls mention-list custom-scrollbar">
```
Comment (crui3er) [inline src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionList.razor:12]: Can we apply 'custom-scrollbar' to 'mention-list'?

File: src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionList.razor [hunk 2]
```diff
@@ -1,36 +1,50 @@
...
     @foreach (var mention in mentions) {
         var searchMatch = mention.SearchMatch;
         var title = searchMatch.Text;
-        var cls = mention == selection ? "bg-04" : "";
-        <div class="@cls px-1 rounded hover:cursor-pointer"
+        var cls = mention == selection ? "bg-mention-list-hover" : "";
+        <div class="@cls hover:cursor-pointer mention-item"
```
Comment (crui3er) [inline src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionList.razor:17]: Can we apply 'cursor-pointer' to 'mention-item'? 'hover' prefix seems not needed.

File: src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionList.razor [hunk 3]
```diff
@@ -1,36 +1,50 @@
...
+                        }
+                    </div>
+                </div>
             }
         </div>
     }
 </div>
 
 @code {
+    [Inject] private IJSRuntime JS { get; init; } = null!;
```
Comment (crui3er) [inline src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionList.razor:41]: '[Inject] private IJSRuntime' should be placed before parameters, but after fields declaration.

File: src/dotnet/Chat.UI.Blazor/Components/MentionList/mention-list.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        this.mentionList = mentionList;
+        this.blazorRef = blazorRef;
+        this.mentionListObserver = new MutationObserver(this.scrollAtCurrentItem);
+        this.mentionListObserver.observe(this.mentionList, {
+            attributes: true,
+            childList: true,
+            subtree: true,
+        })
+        const rect = this.mentionList.getBoundingClientRect();
+        this.listTop = rect.top;
```
Comment (crui3er) [inline src/dotnet/Chat.UI.Blazor/Components/MentionList/mention-list.ts:24]: At creation moment 'hidden' class is applied to mentionList. Are values of listTop and listBottom not equal at that time?

File: src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionList.razor [hunk 5]
```diff
@@ -1,36 +1,50 @@
...
 }
 
-<div class="@Class mb-1 p-1 border border-mention rounded-md overflow-y-auto max-h-96 w-full">
+<div @ref="Ref" class="@Class @listCls mention-list custom-scrollbar">
     @foreach (var mention in mentions) {
         var searchMatch = mention.SearchMatch;
         var title = searchMatch.Text;
-        var cls = mention == selection ? "bg-04" : "";
-        <div class="@cls px-1 rounded hover:cursor-pointer"
+        var cls = mention == selection ? "bg-mention-list-hover" : "";
```
Comment (crui3er) [inline src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionList.razor:16]: I think better to add class 'selected' to a list item when it's selected.
In css define selector:
.mention-item.selected {
  @apply bg-mention-list-hover;
}

Will it work? What do you think?

File: src/nodejs/styles/colors.css [hunk 6]
```diff
@@ -78,6 +78,7 @@
     --message-hover: #F8FBFE;
     --mention: #fed7aa;
     --mention-hover: #fb923c;
+    --mention-list-hover: #EDF2FA;
```
Comment (crui3er) [inline src/nodejs/styles/colors.css:81]: May be better call it '--mention-list-item-selected'

File: src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionList.razor (hunk 1 above)
Comment (iqmulator) [reply src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionList.razor:12]: its use in css causes errors

File: src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionList.razor (hunk 2 above)
Comment (iqmulator) [reply src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionList.razor:17]: grrrr...... yes, i forgot to do it.

File: src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionList.razor (hunk 3 above)
Comment (iqmulator) [reply src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionList.razor:41]: ok

File: src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionList.razor (hunk 5 above)
Comment (iqmulator) [reply src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionList.razor:16]: I think you're right

File: src/nodejs/styles/colors.css (hunk 6 above)
Comment (iqmulator) [reply src/nodejs/styles/colors.css:81]: May be. I will rename it.

**Review by crui3er — APPROVED** [review]

**Pull request merged**